game-core/src/main/java/games/strategy/triplea/ui/menubar/ViewMenu.java

```diff
@@ -435,72 +431,6 @@ private void addLockMap() {
     add(lockMapBox);
   }
 
-  private void addUnitNationDrawMenu() {
-    final JMenu unitSizeMenu = new JMenu();
-    unitSizeMenu.setMnemonic(KeyEvent.VK_N);
-    unitSizeMenu.setText("Flag Display Mode");
-
-    final Preferences prefs = Preferences.userNodeForPackage(getClass());
-    final UnitsDrawer.UnitFlagDrawMode setting =
-        Enum.valueOf(
-            UnitsDrawer.UnitFlagDrawMode.class,
-            prefs.get(
-                UnitsDrawer.PreferenceKeys.DRAW_MODE.name(),
-                UnitsDrawer.UnitFlagDrawMode.NEXT_TO.toString()));
-    UnitsDrawer.setUnitFlagDrawMode(setting, prefs);
-    UnitsDrawer.enabledFlags =
-        prefs.getBoolean(
-            UnitsDrawer.PreferenceKeys.DRAWING_ENABLED.name(), UnitsDrawer.enabledFlags);
-
-    final JCheckBoxMenuItem toggleFlags = new JCheckBoxMenuItem("Show Unit Flags");
-    toggleFlags.setSelected(UnitsDrawer.enabledFlags);
-    toggleFlags.addActionListener(
-        e -> {
-          UnitsDrawer.enabledFlags = toggleFlags.isSelected();
-          prefs.putBoolean(
-              UnitsDrawer.PreferenceKeys.DRAWING_ENABLED.name(), toggleFlags.isSelected());
-          frame.getMapPanel().resetMap();
-        });
-    unitSizeMenu.add(toggleFlags);
-
-    final ButtonGroup unitFlagSettingGroup = new ButtonGroup();
-    unitSizeMenu.add(
-        newFlagDrawModeRadioButtonItem(
-            "Small", unitFlagSettingGroup, UnitsDrawer.UnitFlagDrawMode.NEXT_TO, setting, prefs));
-    unitSizeMenu.add(
-        newFlagDrawModeRadioButtonItem(
-            "Large", unitFlagSettingGroup, UnitsDrawer.UnitFlagDrawMode.BELOW, setting, prefs));
-    add(unitSizeMenu);
-  }
-
-  private JRadioButtonMenuItem newFlagDrawModeRadioButtonItem(
-      final String text,
-      final ButtonGroup group,
-      final UnitsDrawer.UnitFlagDrawMode drawMode,
-      final UnitsDrawer.UnitFlagDrawMode setting,
-      final Preferences prefs) {
-    return newRadioButtonItem(
-        text,
-        group,
-        e -> {
-          UnitsDrawer.setUnitFlagDrawMode(drawMode, prefs);
-          frame.getMapPanel().resetMap();
-        },
-        setting == drawMode);
-  }
-
-  private static JRadioButtonMenuItem newRadioButtonItem(
-      final String text,
-      final ButtonGroup group,
-      final ActionListener action,
-      final boolean selected) {
-    final JRadioButtonMenuItem buttonItem = new JRadioButtonMenuItem(text);
-    buttonItem.addActionListener(action);
-    buttonItem.setSelected(selected);
-    group.add(buttonItem);
-    return buttonItem;
-  }
-
```

Comment:
I would like to keep the unit flags as a view menu option here as they fit well with the other unit UI settings such as unit size, show units, etc.

Reply:
Okay, updated: 7ff1c8e80
What's your long term vision @ron-murhammer for the options and controls of this feature?

Reply:
Well for unit flags in particular, I'd like them to be per map settings like unit size. In terms of location, they are more of a UI setting and would like to eventually refactor the top settings toolbar into more of a modal window like most games have.